src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1627 @@
+            auto const page = (*sleCID)[sfOwnerNode];
+            if (!psb.dirRemove(
+                    keylet::ownerDir(cidOwner), page, sleCID->key(), true))
+            {
+                JLOG(j.fatal())
+                    << "Unable to delete xchain seq number from owner.";
+                return tefBAD_LEDGER;
+            }
+
+            // Remove the sequence number from the ledger
+            psb.erase(sleCID);
+
+            adjustOwnerCount(psb, sleOwner, -1, j);
+        }
+    }
+
+    if (!rewardAccounts.empty())
+    {
+        // distribute the reward pool
+        // if the transfer failed, distribute the pool for "OnTransferFail"
+        // cases (the attesters did their job)
+        STAmount const share = [&] {
+            STAmount const den{rewardAccounts.size()};
+            return divide(rewardPool, den, rewardPool.issue());
+        }();
+        STAmount distributed = rewardPool.zeroed();
+        for (auto const& ra : rewardAccounts)
+        {
+            auto const thTer = transferHelper(
+                psb,
+                rewardPoolSrc,
+                ra,
+                /*dstTag*/ std::nullopt,
+                // claim owner is not relevant to distributing rewards
+                /*claimOwner*/ std::nullopt,
+                share,
+                TransferHelperCanCreateDst::no,
+                j);
+
+            if (thTer == tecINSUFFICIENT_FUNDS || thTer == tecINTERNAL)
+                return thTer;
+
+            if (isTesSuccess(thTer))
+                distributed += share;
+
+            // let txn succeed if error distributing rewards (other than
+            // inability to pay)
+        }
+
+        if (distributed > rewardPool)
+            return tecINTERNAL;
+    }
+
+    return thTer;
+}
+
+/** Get signers list corresponding to the account that owns the bridge
+
+    @param view View to read the signer's list from.
+    @param sleB Sle of the bridge.
+    @param j Log
+
+    @return map of the signer's list (AccountIDs and weights), the quorum, and
+            error code
+*/
+std::tuple<std::unordered_map<AccountID, std::uint32_t>, std::uint32_t, TER>
+getSignersListAndQuorum(ApplyView& view, SLE const& sleB, beast::Journal j)
+{
+    std::unordered_map<AccountID, std::uint32_t> r;
+    std::uint32_t q = std::numeric_limits<std::uint32_t>::max();
+
+    auto const sleS = view.read(keylet::signers(sleB[sfAccount]));
+    if (!sleS)
+        return {r, q, tecXCHAIN_NO_SIGNERS_LIST};
+    q = (*sleS)[sfSignerQuorum];
+
+    auto const accountSigners = SignerEntries::deserialize(*sleS, j, "ledger");
+
+    if (!accountSigners)
+    {
+        return {r, q, tecINTERNAL};
+    }
+
+    for (auto const& as : *accountSigners)
+    {
+        r[as.account] = as.weight;
+    }
+
+    return {std::move(r), q, tesSUCCESS};
+};
+}  // namespace
+//------------------------------------------------------------------------------
+
+NotTEC
+BridgeCreate::preflight(PreflightContext const& ctx)
```

Comment:
Is this supposed to be `XChainCreateBridge`?

Reply:
It probably would be better for consistency with the other xchain transaction class names.

Reply:
Fixed in 91de824b2 [FOLD] Address PR feedback 1: